soc/arm/nordic_nrf/nrf52/Kconfig.soc

```diff
@@ -222,3 +222,32 @@ config NRF_ENABLE_ICACHE
 	bool "Enable the instruction cache (I-Cache)"
 	depends on SOC_NRF52832 || SOC_NRF52840
 	default y
+
+if SOC_SERIES_NRF52X
```

Comment:
These options are available only for nRF52840, so they should not be presented for whole nRF52 series.